scripts/mkcloud

```diff
@@ -1431,11 +1431,19 @@ function expand_steps()
             if [ $onecmd = $cmd ] ; then
                 found=1
                 case "$cmd" in
+                    setupcompute|instcompute)
+                        corrected=${cmd/compute/nodes}
+                        echo "WARNING: the '$cmd' step is deprecated; use '$corrected' instead." >&2
```

Comment:
Actually the `complain` would not exit the script; it would merely exit the `expand_steps` function which is being run in a subshell.  This highlights more reasons why shell really sucks:

- there is no way for functions to return anything except an integer (`$?`)
- this forces callers to use subshells to capture the output of functions returning strings by echoing (and this does not even work for more complex data structures) 
- there is no way to bubble up exceptions through the call stack and then optionally catch them at a higher point